HeterogeneousCore/AlpakaTest/plugins/alpaka/TestAlpakaGlobalProducer.cc

```diff
@@ -21,7 +21,10 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
   class TestAlpakaGlobalProducer : public global::EDProducer<> {
   public:
     TestAlpakaGlobalProducer(edm::ParameterSet const& config)
-        : esToken_(esConsumes()), deviceToken_{produces()}, size_{config.getParameter<int32_t>("size")} {}
+        : esToken_(esConsumes()),
+          deviceToken_{produces()},
+          size_{config.getParameter<edm::ParameterSet>("size").getParameter<int32_t>(
+              EDM_STRINGIZE(ALPAKA_ACCELERATOR_NAMESPACE))} {}
```

Comment:
IIUC, this change is independent from the rest of the PR, and shows a possible way to implement per-backend values for a given parameter ?

Reply:
Correct. It follows the approach first shown in https://github.com/cms-sw/cmssw/pull/40403/files#diff-314fd906d80cd2c6870272476c610a35fcd04727a901b349ea1a1414e3c79b77 . I consider these particular codes more as proof-of-concept that have room for simplification (as also noted in a code comment in #40403).